### PR TITLE
#2: Create a pr check workflow

### PR DESCRIPTION
**Pull request by ShashankKasera**
PR check workflow added in which the following things are there.
~  Gradle Cache
~ Run unit tests
~ Run Detekt

### CODE REVIEW

File: .github/workflows/Pr_chack_workflow.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
+name: Pull request CI checks
+
+on:
+  # Triggers the workflow when code is pushed on master branch
```
Comment (mayankkasera) [inline .github/workflows/Pr_chack_workflow.yml:4]: @ShashankKasera correct the comment

File: .github/workflows/Pr_chack_workflow.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      # run unit tests
+      - name: change wrapper permission
+        id: change_wrapper_permission
+        run: chmod +x ./gradlew
+
+      # run unit tests
+      - name: Run unit tests
+        id: run_unit_tests
+        run: ./gradlew testDebugUnitTest
+
```
Comment (mayankkasera) [inline .github/workflows/Pr_chack_workflow.yml:51]: Add comment here as Run Detekt

File: .github/workflows/Pr_chack_workflow.yml (hunk 1 above)
Comment (ShashankKasera) [reply .github/workflows/Pr_chack_workflow.yml]: done

File: .github/workflows/Pr_chack_workflow.yml (hunk 2 above)
Comment (ShashankKasera) [reply .github/workflows/Pr_chack_workflow.yml:51]: comment edited

**Review by mayankkasera — APPROVED** [review]

**Pull request merged**